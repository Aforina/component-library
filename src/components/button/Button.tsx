import { Button as AntdButton } from 'antd'

export interface ButtonProps extends React.ComponentProps<typeof AntdButton> {}

const Button = (props: ButtonProps) => {
  return (
    <AntdButton
      className='bg-[#87CEEB] font-semibold hover:text-black shadow-[4px_4px_0px_0px_rgba(0,0,0,1)] text-[24px] rounded-0 border-2 border-black px-[32px] py-[26px] flex items-center justify-center'
      {...props}
    >
      {props.children}
    </AntdButton>
  )
}

export default Button
