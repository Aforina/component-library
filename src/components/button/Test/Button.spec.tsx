import { describe, it, expect } from 'vitest'

import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import Button from '../Button'

describe('Button Component Behaviour', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<Button>Button example test</Button>)
    expect(asFragment()).toMatchSnapshot()
  })
})
