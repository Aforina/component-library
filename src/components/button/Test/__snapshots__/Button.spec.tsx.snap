// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component Behaviour > matches the snapshot 1`] = `
<DocumentFragment>
  <button
    class="ant-btn css-dev-only-do-not-override-qgg3xn ant-btn-default bg-[#87CEEB] font-semibold hover:text-black shadow-[4px_4px_0px_0px_rgba(0,0,0,1)] text-[24px] rounded-0 border-2 border-black px-[32px] py-[26px] flex items-center justify-center"
    type="button"
  >
    <span>
      Button example test
    </span>
  </button>
</DocumentFragment>
`;
